app/main.py

```diff
@@ -1,3 +1,52 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: ("Distance", int, float)) -> None:
+        if not isinstance(other, Distance):
+            return Distance(self.km + other)
+        return Distance(self.km + other.km)
+
+    def __iadd__(self, other: ("Distance", int, float)) -> None:
+        if not isinstance(other, Distance):
+            self.km += other
+        else:
+            self.km += other.km
+        return self
+
+    def __mul__(self, other: (int, float)) -> None:
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: (int, float)) -> None:
+        return Distance(round((self.km / other), 2))
+
+    def __lt__(self, other: ("Distance", int, float)) -> bool:
+        if not isinstance(other, Distance):
+            return self.km < other
+        return self.km < other.km
+
+    def __gt__(self, other: ("Distance", int, float)) -> bool:
```

Comment:
The type hint for `other` should be `Union[Distance, int, float]` instead of `("Distance", int, float)` to properly indicate that `other` can be a `Distance` object, an `int`, or a `float`.